kolibri/core/lessons/viewsets.py

```diff
@@ -97,3 +101,26 @@ def annotate_queryset(self, queryset):
         return annotate_array_aggregate(
             queryset, lesson_assignment_collections="lesson_assignments__collection"
         )
+
+    @action(detail=False)
+    def size(self, request, **kwargs):
+        lessons = self.get_queryset()
+        if lessons:
+            lessons_nodes_list = [{lesson.id: lesson.resources} for lesson in lessons]
+            lessons_sizes_dict = {}
+
+            for lesson_node_item in lessons_nodes_list:
+                current_lesson_id = list(lesson_node_item.keys())[0]
+                current_lesson_resources_list = list(lesson_node_item.values())[0]
+                resource_nodes = []
+
+                for resource in current_lesson_resources_list:
+                    resource_nodes.append(
+                        ContentNode.objects.filter(id=resource["contentnode_id"])
+                    )
+
+                lessons_sizes_dict[current_lesson_id] = total_file_size(
+                    resource_nodes[0]
```

Comment:
I think you probably want all the `resource_nodes` here, otherwise you'll just be getting the size of the first one!
```
resource_nodes
```